### PR TITLE
Issue/442 do not ignore node modules

### PR DESCRIPTION
**Pull request by FloLey**
new modules where ignore by .vscodeignore.

issue #442

### CODE REVIEW

**Review by sanderr — CHANGES_REQUESTED** [review]
Could you also fix the pipeline (`.ci/Jenkinsfile-extension-vsce`)? `npm ci` should be run after checking out the appropriate branch.

**Comment by sanderr** [comment]
Oh, and it should install with `--omit dev`

**Comment by sanderr** [comment]
Hmm, by adding `--omit dev` `vsce` will not be installed. I'm not sure how to install that separately while respecting the constraints in `package.json` and `package-lock.json`. That one should probably remain in the vscodeignore too.

**Review by Stijnkool — APPROVED** [review]

File: .ci/Jenkinsfile-extension-vsce [hunk 1]
```diff
@@ -168,7 +165,13 @@ pipeline {
             }
             steps {
                 dir('vscode-inmanta') {
-                    sh '$(npm bin)/vsce package'
+                    sh '''
+                        # install the extension for its vsce dependency, required for building and publishing the extension
+                        # and populate vscodeignore with exceptions to node_modules of the needed modules.
+                        npm ci --omit dev && find node_modules/ -mindepth 1 -maxdepth 1 -exec echo '!{}/**' \; >> .vscodeignore && npm i
```
Comment (sanderr) [inline .ci/Jenkinsfile-extension-vsce:171]: I would split them up to make this more readable
```suggestion
                        npm ci --omit dev
                        find node_modules/ -mindepth 1 -maxdepth 1 -exec echo '!{}/**' \; >> .vscodeignore
                        npm i
```

File: .vscodeignore [hunk 2]
```diff
@@ -10,11 +10,4 @@ tsconfig.json
...
 **/*.map
-
-node_modules/**
-!node_modules/vscode-jsonrpc/**
-!node_modules/vscode-languageclient/**
-!node_modules/vscode-languageserver-protocol/**
-!node_modules/vscode-languageserver-types/**
-!node_modules/semver/**
-!node_modules/get-port/**
+!node_modules//**
```
Comment (sanderr) [inline .vscodeignore:13]:
```suggestion
node_modules/**
```
Is this what you meant? If not, could you elaborate?

**Review by sanderr — CHANGES_REQUESTED** [review]

**Review by sanderr — APPROVED** [review]

**Pull request merged**